src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/MermaidDumperCommandPass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\DependencyInjection\Compiler;
+
+use Symfony\Bundle\SecurityBundle\Command\SecurityRoleHierarchyDumpCommand;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Registers the SecurityRoleHierarchyDumpCommand only if RoleHierarchy is available.
+ *
+ * @author Damien Fernandes <damien.fernandes24@gmail.com>
+ */
+class MermaidDumperCommandPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (!$container->hasDefinition('security.role_hierarchy')) {
+            return;
+        }
+
+        $definition = new Definition(SecurityRoleHierarchyDumpCommand::class);
```

Comment:
The minimum version of `symfony/security-core` in the dependencies of `symfony/security-bundle` needs to be bumped to 7.4 to ensure that the dependencies of this command exist.
Otherwise, we would have to deal with compatibility with older versions of the package (not worth it in this case, as 7.3 will be EOL 2 months after the release of 7.4)